src/pip/_internal/download.py

```diff
@@ -101,6 +101,14 @@
         )
 
 
+try:
+    from pip._vendor.requests_kerberos import HTTPKerberosAuth
+    from pip._vendor.requests_kerberos import kerberos_ as ik
+    _KERBEROS_AVAILABLE = True
```

Comment:
"If py/winkerberos is not installed, the behaviour is the same as before": requests_kerberos on its own will not make kerberos work, one has to have either py/winkerberos installed (which we cannot vendor because they are compiled/platform dependent?). requests_kerberos in that case will fail with an ImportError.